fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,227 @@
+                log.fine("Publishing NATS notification event to channel '" + channelName + "',\nmessage: '" + jsonString + "'.");
+            }
+            
+            sc.publish("FHIRNotificationEvent", jsonString.getBytes(), acb);
+    
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Published NATS notification event to channel '" + channelName + "'");
+            }
+        } catch (Throwable e) {
+            String msg = buildNotificationErrorMessage(channelName, (jsonString == null ? "<null>" : jsonString));
+            log.log(Level.SEVERE, msg , e);
+            throw new FHIRNotificationException(msg, e);
+        } finally {
+            log.exiting(this.getClass().getName(), "notify");
+        }
+    }
+    
+    /**
+     * Builds a formatted error message to indicate a notification publication failure.
+     */
+    private String buildNotificationErrorMessage(String channelName, String notificationEvent) {
+        return String.format("NATS publication failure; channel '%s'\nNotification event: '%s'\n.", channelName, notificationEvent);
+    }
+}
+
+/* 
+* Modified from original NATS documentation @ https://docs.nats.io/developing-with-nats/security/tls
+*
+* openssl is used to generate a pkcs12 file (.p12) from client-cert.pem and client-key.pem.
+* The resulting file is then imported into a java keystore using keytool which is part of java jdk.
+* keytool is also used to import the CA certificate rootCA.pem into the truststore.  
+*/
+class SSLUtils extends FHIRNotificationNATSPublisher {
+
+    static KeyStore loadKeystore(String path, String password) throws Exception {
+        KeyStore store = KeyStore.getInstance("PKCS12");
+
+        try(BufferedInputStream in = new BufferedInputStream(new FileInputStream(path));) {
+            store.load(in, password.toCharArray());
+        }
+        return store;
+    }
+
+    static KeyManager[] createTestKeyManagers(Properties tlsProps) throws Exception {
+        KeyStore store = loadKeystore(tlsProps.getProperty("keystore"), tlsProps.getProperty("keystore-pw"));
+        KeyManagerFactory factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        factory.init(store, tlsProps.getProperty("keystore-pw").toCharArray());
+        return factory.getKeyManagers();
+    }
+
+    static TrustManager[] createTestTrustManagers(Properties tlsProps) throws Exception {
+        KeyStore store = loadKeystore(tlsProps.getProperty("truststore"), tlsProps.getProperty("truststore-pw"));
+        TrustManagerFactory factory = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        factory.init(store);
+        return factory.getTrustManagers();
+    }
+
+    static SSLContext createSSLContext(Properties tlsProps) throws Exception {
+        SSLContext ctx = SSLContext.getInstance("TLSv1.2");
+        ctx.init(createTestKeyManagers(tlsProps), createTestTrustManagers(tlsProps), new SecureRandom());
```

Comment:
Is this really a test trust managers? 
It's not really test, I suggest renaming the test methods in this class, and refactoring into a single class.

Reply:
I can do that when the code is refactored to make those methods private methods of the publisher class.